geoips/interfaces/module_based/output_checkers.py

```diff
@@ -0,0 +1,73 @@
+# # # Distribution Statement A. Approved for public release. Distribution unlimited.
+# # #
+# # # Author:
+# # # Naval Research Laboratory, Marine Meteorology Division
+# # #
+# # # This program is free software: you can redistribute it and/or modify it under
+# # # the terms of the NRLMMD License included with this program. This program is
+# # # distributed WITHOUT ANY WARRANTY; without even the implied warranty of
+# # # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the included license
+# # # for more details. If you did not receive the license, for more information see:
+# # # https://github.com/U-S-NRL-Marine-Meteorology-Division/
+
+"""Output Checkers interface module."""
+
+from geoips.interfaces.base import BaseModuleInterface, BaseModulePlugin
+import logging
+
+LOG = logging.getLogger(__name__)
+
+
+class OutputCheckersBase(BaseModulePlugin):
+    """Output Checkers routines to apply when comparing data outputs."""
+
+
+class OutputCheckersInterface(BaseModuleInterface):
+    """Output Checkers routines to apply when comparing data outputs."""
+
+    name = "output_checkers"
+    required_args = {"standard": {}}
+    required_kwargs = {"standard": {}}
+    # required_args = {
+    #     "standard": ["fname", "output_product", "compare_product"],
+    #     "print_gunzip": ["fobj", "gunzip_fname"],
+    #     "print_gzip": ["fobj", "gzip_fname"],
+    # }
+    # required_kwargs = {"standard": {}}
+    # allowable_kwargs = {
+    #     "test_product": ["goodcomps", "badcomps", "compare_strings"],
+    #     "out_diffname": ["ext", "flag"],
+    #     "compare_outputs": ["test_product_func"],
+    # }
+    plugin_class = OutputCheckersBase
+
+    def get_checker(self, filename):
+        """Find and Return the correct output checker for the provided filename."""
+        from geoips.plugins.modules.output_checkers import (
+            geotiff_checker,
+            gz_checker,
+            image_checker,
+            netcdf_checker,
+            text_checker,
+        )
+
+        output_checkers = [
+            geotiff_checker,
+            image_checker,
+            gz_checker,
+            netcdf_checker,
+            text_checker,
+        ]
+        correct_checker = None
```

Comment:
Shouldn't these be automatically discovered, without explicitly importing each one?  We want to be able to add new output checkers to other repos, without referencing them here?

Reply:
You're right. That was a brain fart on my side! I'll change that now.